App design candidate

nfpirus opened the pull request:
Необходимо утвердить предложенный вариант реализации архитектуры приложения

nfpirus committed:
upload pre

upload pre

list64 committed:
VS project

Поместил в проект Visual Studio 2015

list64 committed:
add methods

добавил методы

nihlete commented on src/Render.ts:
```diff
@@ -0,0 +1,32 @@
+
+        this._ctx = this._canvas.getContext('2d');
+
+        paper.install(window);
```
`paper.install` устанавливает в глобальную область видимости всё содержимое библиотеки. При разработке в любом случае используется обращение к элементам библиотеки в формате:

```
paper.Path({});
paper.Group({});
```

Пространство имен библиотеки большое, а устанавливать его ни к чему

nihlete commented on src/Stage.ts:
```diff
@@ -1,25 +1,29 @@
-        const myLine: any = new MyLine(40, 90, 90, 40);
-
-
+    private deliteShape(shape: IShape): void {
```
Наверно ты хотел написать delete

nihlete commented on src/Stage.ts:
```diff
@@ -1,25 +1,29 @@
+    private deliteShape(shape: IShape): void {
+        const item: number = this._shapes.indexOf(shape);
+        if (item) {
+            this._shapes.splice(item, 1);
```
Удаляя объект из массива, ты не удаляешь его с канваса. Для удаления с канваса используют `.remove();`.
здесь надо написать так:

```
this._shapes.renderObject.remove();
this._shapes.splice(item, 1);
```

list64 committed:
render refactor

- правки по комментариям
- рефакторинг Render и RenderApi упростил вызов функций

nihlete commented on src/IShape.ts:
```diff
@@ -0,0 +1,7 @@
+interface IShape {
+    type: number;
+    coordinates: ICoordinates;
```
Два варианта модели объекта: писать все в натуральных координатах и рендерить с масштабом. При изменении масштаба все на основании реальных координат пересчитывается. 

Второй вариант использовать методы библиотеки для масштабирования и рисовать в пикселях. Изменение масштаба вида соответствует изменению масштаба объектов. 
Мне больше нравится первый вариант, таким образом отделим модель сцены от экрана.

P.s. А еще я не понял разницу между `coord`и `coordDraw`

nfpirus replied:
coord - координаты в системе координат сцены
coordDraw - координаты для отрисовки Canvas холста
В данный момент если я правильно понял то что написал реализован первый метод, и как бы coordDraw можно не хранить, однако это необходимо для абстрагирования от RenderApi

list64 committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into appDesignCandidate-

list64 committed:
добавил исключение
